pkg/translate/translateValue.go

```diff
@@ -0,0 +1,554 @@
+		minPath := newP + ".autoscaleMin"
+		asMin, found, err := name.GetFromTreePath(valueTree, util.ToYAMLPath(minPath))
+		if found && err == nil {
+			if err := tpath.WriteNode(cpSpecTree, util.ToYAMLPath(outPath+".minReplicas"), asMin); err != nil {
+				return err
+			}
+		}
+		if _, err := DeleteFromTree(valueTree, util.ToYAMLPath(minPath), util.ToYAMLPath(minPath)); err != nil {
+			return err
+		}
+		maxPath := newP + ".autoscaleMax"
+		asMax, found, err := name.GetFromTreePath(valueTree, util.ToYAMLPath(maxPath))
+		if found && err == nil {
+			log.Infof("path has value in helm Value.yaml tree, mapping to output path %s", outPath)
+			if err := tpath.WriteNode(cpSpecTree, util.ToYAMLPath(outPath+".maxReplicas"), asMax); err != nil {
+				return err
+			}
+		}
+		if _, err := DeleteFromTree(valueTree, util.ToYAMLPath(maxPath), util.ToYAMLPath(maxPath)); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// translateEnv translates env value from helm values.yaml tree
+func translateEnv(outPath string, value interface{}, cpSpecTree map[string]interface{}) error {
+	envMap, ok := value.(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("expect env node type to be map[string]interface{} but got: %T", value)
+	}
+	outEnv := make([]map[string]interface{}, len(envMap))
+	cnt := 0
+	for k, v := range envMap {
+		outEnv[cnt] = make(map[string]interface{})
+		outEnv[cnt]["name"] = k
+		outEnv[cnt]["value"] = v
+		cnt++
+	}
+	log.Infof("path has value in helm Value.yaml tree, mapping to output path %s", outPath)
+	if err := tpath.WriteNode(cpSpecTree, util.ToYAMLPath(outPath), outEnv); err != nil {
+		return err
+	}
+	return nil
+}
+
+// translateAffinity translates Affinity related configurations from helm values.yaml tree
+func translateAffinity(outPath string, value interface{}, cpSpecTree map[string]interface{}) error {
+	affinityNode, ok := value.([]interface{})
+	if !ok {
+		return fmt.Errorf("expect affinity node type to be []interface{} but got: %T", affinityNode)
+	}
+	log.Infof("path has value in helm Value.yaml tree, mapping to output path %s", affinityNode)
+	if err := tpath.WriteNode(cpSpecTree, util.ToYAMLPath(outPath), affinityNode); err != nil {
+		return err
+	}
+	return nil
+}
+
+// translateK8sTree is internal method for translating K8s configurations from value.yaml tree.
+func (t *ValueYAMLTranslator) translateK8sTree(valueTree map[string]interface{},
+	cpSpecTree map[string]interface{}) error {
+	for inPath, v := range t.KubernetesMapping {
+		log.Infof("Checking for k8s path %s in helm Value.yaml tree", inPath)
+		m, found, err := name.GetFromTreePath(valueTree, util.ToYAMLPath(inPath))
+		if err != nil {
+			return err
+		}
+		if !found {
+			log.Infof("path %s not found in helm Value.yaml tree, skip mapping.", inPath)
+			continue
+		}
+		path := util.PathFromString(inPath)
+		k8sSettingName := ""
+		if len(path) != 0 {
+			k8sSettingName = path[len(path)-1]
+		}
+		if k8sSettingName == "autoscaleEnabled" {
```

Comment:
better to write this as
switch k8sSettingName {
case "autoscaleEnabled":
...

Reply:
done